lib/http/HttpServer.ts

```diff
@@ -67,15 +67,22 @@ class HttpServer {
       res.writeHead(200, { 'Content-Type': 'application/json' });
       res.end(JSON.stringify(resJson));
     } catch (err) {
-      if (err.code === errorCodes.INVALID_ARGUMENT) {
-        res.writeHead(400);
-        res.end(err.message);
-      } else if (err.code === swapErrorCodes.PAYMENT_HASH_NOT_FOUND) {
-        res.writeHead(404);
-        res.end(err.message);
-      } else {
-        res.writeHead(500);
-        res.end(JSON.stringify(err));
+      switch (err.code) {
+        case errorCodes.INVALID_ARGUMENT:
+          res.writeHead(400);
+          res.end(err.message);
+          break;
+        case swapErrorCodes.INVALID_RESOLVE_REQUEST:
```

Comment:
This can just flow with the previous case like so we don't need to write the code twice:

```javascript
        case errorCodes.INVALID_ARGUMENT:
        case swapErrorCodes.INVALID_RESOLVE_REQUEST:
          res.writeHead(400);
          res.end(err.message);
          break;
```